bin/expo-setup.js

```diff
@@ -1,15 +1,23 @@
-// Enable Expo non-default options for performance:
+// Enable Expo non-default options for performance and consistency:
 //
-// 1. app.json - Enable New Architecture for iOS and Android
-//    - https://docs.expo.dev/guides/new-architecture/
+// 1. Convert app.json to app.config.ts for consistent TS language and dynamic config
+//    - https://docs.expo.dev/workflow/configuration/#dynamic-configuration
+//
+//    TODO: Remove this if `create-expo-app` generates `app.config.ts` in future
+//    - https://github.com/expo/expo/issues/34357
 // 2. .env.development, .env.production, eas.json - Enable the new Metro resolver available starting in Expo SDK 51
 //    - https://github.com/EvanBacon/pillar-valley/commit/ede321ef7addc67e4047624aedb3e92af3cb5060
 //    - https://archive.ph/MG03E
 //
 // TODO: Remove when Expo enables New Architecture and new Metro resolver by default
-import { readFile, writeFile } from 'node:fs/promises';
+import { execSync } from 'node:child_process';
+import { readFile, unlink, writeFile } from 'node:fs/promises';
 import isPlainObject from 'is-plain-obj';
 
+execSync('pnpm add --save-dev prettier', { stdio: 'inherit' });
```

Comment:
- let's stick with async APIs, instead of the `*Sync` functions
- because this is unusual (using pnpm to install via `exec`, dynamic import below) let's add a comment above this why it's better (mention colocation and maybe ease of deleting the code later when it's not needed anymore, add a link to the Expo tracking issue/PR)

Reply:
1. used `promisify(exec)` now to install Prettier [f8cdd74](https://github.com/upleveled/eslint-config-upleveled/pull/462/commits/f8cdd745c9aa1f37e8664621b2e91d21248f4cd3)
2. added a comment [c23acfc](https://github.com/upleveled/eslint-config-upleveled/pull/462/commits/c23acfc5dde69ea2c63acab9fb988cdc21c89356)